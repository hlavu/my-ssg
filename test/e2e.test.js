const run = require("./run");

describe("end-to-end integration", () => {
  test("prints help message when --help option is passed", async () => {
    const { stderr, stdout, exitCode } = await run("--help");

    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints version when --version option is passed", async () => {
    const { stderr, stdout, exitCode } = await run("--version");

    expect(exitCode).toBe(0);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });
});
