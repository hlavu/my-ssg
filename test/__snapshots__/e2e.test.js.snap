// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end integration prints help message when --help option is passed 1`] = `
"Usage: index.js --input <filename>  [-s <css-link>] [-l <lang-code>]

Options:
  -i, --input       .txt or .md file name                                [array]
  -c, --config      configuration file                     [array] [default: \\"\\"]
  -a, --assets      path to assets folder                  [array] [default: \\"\\"]
  -s, --stylesheet  css link                                            [string]
  -l, --lang        language used in HTML                               [string]
  -v, --version     Show version number                                [boolean]
  -h, --help        Show help                                          [boolean]"
`;

exports[`end-to-end integration prints version when --version option is passed 1`] = `"ssg-cli 1.0.0"`;
